application/libraries/Api/Command/V1/SurveyPatch/OpHandlerLanguageSettingsUpdate.php

```diff
@@ -121,14 +108,16 @@ public function handle(OpInterface $op): void
     public function validateOperation(OpInterface $op): array
     {
         $validationData = [];
-        $checkDataEntityId = $this->validateEntityId($op, []);
+        $checkDataEntityId = $this->getSurveyIdFromContext($op);
         $checkDataCollection = $this->validateCollection($op, []);
         if (empty($checkDataEntityId) && empty($checkDataCollection)) {
```

Comment:
This block still needs to be changed. We actually don't care about the entityId anymore. And the data should come in as collection all the time, so 'props can not come as collection if id is set' doesn't make sense anymore.

I would just remove this whole if-block but keep the $validationData = $this->validateCollectionIndex($op, $validationData) 

validateCollectionIndex checks if it's a collection and in this case if the indexes are strings. We don't need anything else